exporter/api/query_handlers.go

```diff
@@ -98,3 +44,56 @@ func HandleUnknownQuery(logger *zap.Logger, nm *NetworkMessage) {
 		Data: []string{fmt.Sprintf("bad request - unknown message type '%s'", nm.Msg.Type)},
 	}
 }
+
+// HandleParticipantsQuery handles TypeParticipants queries.
+func HandleParticipantsQuery(logger *zap.Logger, qbftStorage *storage.QBFTStores, nm *NetworkMessage, domain spectypes.DomainType) {
+	logger.Debug("handles query request",
+		zap.Uint64("from", nm.Msg.Filter.From),
+		zap.Uint64("to", nm.Msg.Filter.To),
+		zap.String("publicKey", nm.Msg.Filter.PublicKey),
+		zap.String("role", nm.Msg.Filter.Role))
+	res := Message{
+		Type:   nm.Msg.Type,
+		Filter: nm.Msg.Filter,
+	}
+	pkRaw, err := hex.DecodeString(nm.Msg.Filter.PublicKey)
+	if err != nil {
+		logger.Warn("failed to decode validator public key", zap.Error(err))
+		res.Data = []string{"internal error - could not read validator key"}
+		nm.Msg = res
+		return
+	}
+
+	beaconRole, err := message.BeaconRoleFromString(nm.Msg.Filter.Role)
+	if err != nil {
+		logger.Warn("failed to parse role", zap.Error(err))
+		res.Data = []string{"role doesn't exist"}
+		nm.Msg = res
+		return
+	}
+	runnerRole := exporter_message.RunnerRole(beaconRole)
+	roleStorage := qbftStorage.Get(runnerRole)
+	if roleStorage == nil {
+		logger.Warn("role storage doesn't exist", fields.Role(spectypes.RunnerRole(runnerRole)))
```

Comment:
`spectypes.RunnerRole(runnerRole)` this cast might error because we don't have all `exporter_message.RunnerRole`'s in `spectypes.RunnerRole`